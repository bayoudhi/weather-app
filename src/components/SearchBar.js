import PropTypes from "prop-types";
import "./SearchBar.css";

function SearchBar({ handleChange }) {
  return (
    <div className="search-box">
      <input
        onChange={(e) => handleChange(e.target.value)}
        type="text"
        className="search-bar"
        placeholder="Search..."
      />
    </div>
  );
}

SearchBar.propTypes = {
  handleChange: PropTypes.func,
};

export default SearchBar;
