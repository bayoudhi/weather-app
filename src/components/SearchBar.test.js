import { render, screen, fireEvent } from "@testing-library/react";
import * as React from "react";
import SearchBar from "./SearchBar";

let handleChange;
let input;
beforeEach(() => {
  handleChange = jest.fn();
  render(<SearchBar handleChange={handleChange} />);
  input = screen.getByRole("textbox");
});

it("text should be empty by default", () => {
  expect(input.value).toBe("");
});

it("should call handleChange 3 times, when text changes 3 times", () => {
  fireEvent.change(input, {
    target: {
      value: "Paris",
    },
  });

  fireEvent.change(input, {
    target: {
      value: "",
    },
  });

  fireEvent.change(input, {
    target: {
      value: "Tunis",
    },
  });

  expect(handleChange).toBeCalledTimes(3);
  expect(handleChange.mock.calls[0][0]).toBe("Paris");
  expect(handleChange.mock.calls[1][0]).toBe("");
  expect(handleChange.mock.calls[2][0]).toBe("Tunis");
});
