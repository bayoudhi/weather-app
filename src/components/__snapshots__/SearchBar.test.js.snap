// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should fire handleChange when text changes 1`] = `
<input
  class="search-bar"
  placeholder="Search..."
  type="text"
/>
`;
